### ISSUE TITLE
0.13.0 and later will hang forever

### ISSUE THREAD

**Issue by adamchalmers** [EmbarkStudios/cargo-deny#474]
**Describe the bug**
When I run `cargo deny check` on my project, it hangs forever. This does not happen in 0.12.0, but happens in every 0.13.x release so far.

**To Reproduce**
I haven't got an open-source reproduction, because the dependency which makes `cargo deny check` hang forever is closed-source. It does use features and a build.rs which generates protobuf stubs using tonic 0.8, and @OrionNebula says

> It uses a bunch of CPU while it's stuck. I was able to pull this stack trace: https://paste.cfdata.org/ViG4W8R17b3LpFjr4cKX Which I think comes from this line: https://docs.rs/krates/latest/src/krates/builder.rs.html#875 (or something in that statement anyway)

**Expected behavior**
cargo deny check should terminate in a few seconds

**Device:**
 - OS: macOS Monterey
 - Architecture: tested on both Intel arm64 and Apple Silicon M1 (aarch64)

**Additional context**
I'm sorry this is so vague, I am trying to reproduce in an open-source way. But I figured I'd open this issue now, even though it's not very helpful, in case anyone else has this problem with an open-source crate.

**Comment by Jake-Shadle** [comment]
Would it be possible to just post the output of `cargo metadata --format-version 1` to a gist?

**Comment by adamchalmers** [comment]
Yeah, here it is: https://gist.github.com/adamchalmers/5fb6e13a7ae63c10de1f260a77730f6c
